### PR TITLE
Calculator app

### PR DESCRIPTION
**Pull request by PuruVJ**

### CODE REVIEW

**Comment by ZhiyueYi** [comment]
Can't wait for this PR to get merged!! Then I can start to build some "Apps" :p

**Comment by PuruVJ** [comment]
> Can't wait for this PR to get merged!! Then I can start to build some "Apps" :p

Yup. I'll figure out all the constraints of making apps (architechture, side components, composition, etc), then the app making should become easy. Let's hope this product gets so good that Apple will want to sue us, but won't cuz we're non-monetised 😉

**Comment by PuruVJ** [comment]
Plus I refactored the window maximize functionality. Its now its own hook which returns a callback.

**Comment by ZhiyueYi** [comment]
> Plus I refactored the window maximize functionality you built yesterday. Its now its own hook which returns a callback.

That's nice 👍 Thanks!

File: src/components/Desktop/Window/Window.tsx [hunk 1]
```diff
@@ -152,15 +87,94 @@ export const Window = ({ appID }: WindowProps) => {
...
+  try {
+    return { x: Number(matched.next().value[0]), y: Number(matched.next().value[0]) };
+  } catch {
+    return { x: 0, y: 0 };
+  }
+}
+
+const useMaximizeWindow = (
+  windowRef: RefObject<WindowRnd>,
+  transitionClearanceRef: RefObject<number>,
```
Comment (ZhiyueYi) [inline src/components/Desktop/Window/Window.tsx:121]: Actually `transitionClearanceRef` is just a simple ref to track the `setTimeout`, which doesn't need to be passed in. I think you can just create this as a `useRef` in `useMaximizeWindow` itself

Comment (PuruVJ) [reply src/components/Desktop/Window/Window.tsx:121]: Great. Will do that

**Comment by ZhiyueYi** [comment]
Do you think it would be a better idea to make a single app into an individual package? Then import the package into the macos app, so that the OS part and application part can be decoupled. Not sure if it's over-engineering, lol

**Comment by PuruVJ** [comment]
> Do you think it would be a better idea to make a single app into an individual package? Then import the package into the macos app, so that the OS part and application part can be decoupled. Not sure if it's over-engineering, lol

I would recommend against that, because its way too complex, not to mention we'd need some kind of registry of sort, which means backend, which means costs to pay, which means monetisation, which means Apple swooping down on me with a billion dollar lawsuit or something lol

Jokes aside, I don't think its a good idea. Remember, I'm not aiming for big apps, just small mundane apps like notes, todo, calculator etc, which are better built into the system.

File: src/components/Desktop/Window/Window.tsx (hunk 1 above)
Comment (PuruVJ) [reply src/components/Desktop/Window/Window.tsx:121]: Hehe, actually only the windowRef was needed as external ref, rest all could be moved into the hook itself

**Comment by PuruVJ** [comment]
So let's merge this. Basic non-functional calculator is done

**Pull request merged**